master/buildbot/status/buildstep.py

```diff
@@ -267,6 +271,10 @@ def addURL(self, name, url, results=None):
         if results is not None:
             self.urls[name] = {'url': url, 'results': results}
 
+    def addArtifactURL(self, name, url, results=None):
+        self.artifactUrls[name] = url
+        if results is not None:
+            self.artifactUrls[name] = {'url': url, 'results': results}
```

Comment:
I am not fun of messing out types like this. In this case with the more advanced operations on JSON objects like reducing or filtering by property - it fails, and we need to convert JSON to a valid one. Let keep JSON consistent. And if results are None, let it be empty.